src/ucp/api/ucp.h

```diff
@@ -1747,6 +1845,74 @@ ucs_status_ptr_t ucp_tag_send_sync_nb(ucp_ep_h ep, const void *buffer, size_t co
 
 /**
  * @ingroup UCP_COMM
+ * @brief Non-blocking stream receive operation.
+ *
+ * This routine receives a data that is described by the local address @a
+ * buffer, size @a count, and @a datatype object on the endpoint @a ep. The
+ * routine is a non-blocking and therefore returns immediately. The receive
+ * operation is considered completed when the message is delivered to the @a
+ * buffer. In order to notify the application about completion of the receive
+ * operation the UCP library will invoke the call-back @a cb when the received
+ * data is in the receive buffer and ready for application access. If the
+ * receive operation cannot be stated the routine returns an error.
```

Comment:
@evgeny-leksikov My point is a bit different. 
you can use some pre-allocated tags for streaming the data since tagged interface provides similar ordering semantics.